### DISCUSSION TITLE
Thumbnail extraction and display

### DISCUSSION THREAD

**Discussion by naari3** [osmoscraft/osmosfeed#75] [Ideas] [answered]
Thanks for the great product!

I would like to display thumbnails on my antennas. like [this](https://naari3.github.io/naari3-feed/) (and [here](https://github.com/naari3/naari3-feed/blob/19aee18c9cfbd3402c9515882e88fef5b82bd405/includes/index.hbs#L35) is index.hbs source).  
This will be displayed in the img tag if there is something in the enclosure tag.

And I would like to support the [this](https://note.com/info/rss) feeds if there is an image in `media:thumbnail`. 

However, osmosfeed doesn't allow you to get arbitrary tags for items?

https://github.com/osmoscraft/osmosfeed/blob/a6bea23459be591b6f41107f57bead83cc1398a6/packages/cli/src/lib/enrich.ts#L16 (It seems that each item is abstracted to simplify it.)

Is there anything I can do about this? thanks!

**Comment by naari3** [comment]
Sorry, I just saw the discuss about v2 https://github.com/osmoscraft/osmosfeed/discussions/70.
As I understand it, will v2 allow free values to be used?
https://github.com/osmoscraft/osmosfeed/blob/aea530445bd1cce8a812a976fc31e179658c0d40/packages/types/json-feed.d.ts#L22

**Reply by chuanqisun** [reply]
The v2 parser engine actually [detects thumbnail](https://github.com/osmoscraft/osmosfeed/blob/aea530445bd1cce8a812a976fc31e179658c0d40/packages/feed-parser/src/lib/parsers.ts#L42) but I forgot to handle `media:thumbnail` as you mentioned. My goal is support all common RSS feed related fields without asking people to add free values. And eventually, I want to support some degree of extension, though I don't have a technical design yet.

I also noticed the popular use case of v1 is becoming more and more different from my original design of v2. People love using v1 for hosting public antennas (like yours), and v2 was designed to be a personal rss reader app. This made me think I should back porting the v2 parser and thumbnail detection logic from v2 to v1 while keeping the hbs templating system. Otherwise, people will have a very difficult time migrating to v2. If I can get this right, you will get thumbnail support in v1, without migration.

p.s. I love https://naari3.github.io/naari3-feed/ . Would you be ok if I implement it as the default index.hbs after v1 can detect thumbnails? I'm sure many people will thank you for the exploration.

**Reply by chuanqisun** [reply]
I have updated the title, from Q&A to Idea. Implementation will be tracked here.

**Reply by naari3** [reply]
> If I can get this right, you will get thumbnail support in v1, without migration.

This is so cool! I hope it works out for you.

> Would you be ok if I implement it as the default index.hbs after v1 can detect thumbnails?

Of course! I love osmosfeed. Would be very happy if [this](https://github.com/naari3/naari3-feed) styles gets used!

**Comment by chuanqisun** [comment, marked as answer]
I just released the thumbnail view in v1.13.0. 
![image](https://user-images.githubusercontent.com/1895289/153739624-caaa6007-2071-4c45-9a38-d0f176f7303c.png)
